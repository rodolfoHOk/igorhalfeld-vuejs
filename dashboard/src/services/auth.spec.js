import mockAxios from 'axios';
import AuthService from './auth';

jest.mock('axios');

describe('Auth service', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return a token when call login', async () => {
    const token = 'a1b.c2d.e3f';

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } });
    });

    const response = await AuthService(mockAxios).login({
      email: 'john.doe@email.com',
      password: '123456',
    });

    expect(response.data).toHaveProperty('token');
    expect(response).toMatchSnapshot();
  });

  it('should return an user when call register', async () => {
    const user = {
      name: 'John Doe',
      email: 'john.doe@email.com',
      password: '1234',
    };

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user });
    });

    const response = await AuthService(mockAxios).register(user);

    expect(response.data).toHaveProperty('name');
    expect(response.data).toHaveProperty('email');
    expect(response.data).toHaveProperty('password');
    expect(response).toMatchSnapshot();
  });

  it('should throw an error when call login and not found user', async () => {
    const errors = { status: 404, statusText: 'Not Found' };

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors });
    });

    const response = await AuthService(mockAxios).login({
      email: 'igor@igor.me',
      password: '123',
    });

    expect(response.errors).toHaveProperty('status');
    expect(response.errors).toHaveProperty('statusText');
  });
});
