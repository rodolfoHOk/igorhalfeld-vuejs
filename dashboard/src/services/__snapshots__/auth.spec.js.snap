// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth service should return a token when call login 1`] = `
Object {
  "data": Object {
    "token": "a1b.c2d.e3f",
  },
  "errors": null,
}
`;

exports[`Auth service should return an user when call register 1`] = `
Object {
  "data": Object {
    "email": "john.doe@email.com",
    "name": "John Doe",
    "password": "1234",
  },
  "errors": null,
}
`;
